Add network throttling and device user agent configs

bmcminn opened the pull request:
Fix #5

**EDIT:**
- :shell: add CLI option `-A` (user agent) with argument validation
- :shell: add CLI option `-Y` (network throttling) with argument validation
- fix `user-agent` string override when using `-A`
- :shell: add CLI option `--landscape` which forces device orientation to landscape
- configure `src/profiles/devices.js` based on Chromium device profile definitions:
  - https://chromium.googlesource.com/chromium/src/+/master/third_party/WebKit/Source/devtools/front_end/emulated_devices/module.json
- configured `src/profiles/networks.js` based on Chromium network speed definitions:
  - https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/devtools/front_end/components/NetworkConditionsSelector.js&q=%22Good%203G%22&sq=package:chromium&dr=C&l=50
- :art: general formatting of scripts
- :clipboard: add commentary to script components

bmcminn committed:
added network throttling and user agent device parameters

shyiko commented:
Good work! Would you mind removing console.log calls and rebase the branch (right now I can't merge it in due to "This branch has conflicts that must be resolved")? 

Also, where does src/profiles/devices.js & src/profiles/networks.js data come from? Thank you.

bmcminn commented:
Alright, finally got the rebase worked out.

Also, the `src/profiles` data was extrapolated from the latest Chrome throttling and UA device configs in dev tools.

bmcminn committed:
added `contributors` field to `package.json`

bmcminn committed:
Merge branch 'master' of https://github.com/shyiko/electron-har

Conflicts:
	package.json

shyiko commented:
I propose we take list of devices from [here](https://chromium.googlesource.com/chromium/src/+/master/third_party/WebKit/Source/devtools/front_end/emulated_devices/module.json). What do you think?

shyiko commented:
Also, download/upload throttling is calculated differently. See [here](https://code.google.com/p/chromium/codesearch#chromium/src/third_party/WebKit/Source/devtools/front_end/components/NetworkConditionsSelector.js&q=%22Good%203G%22&sq=package:chromium&dr=C&l=50).

bmcminn committed:
updated network profile throughput maths to match chromium network profile spec

bmcminn committed:
updated device profiles to latest Chromium source defs; added user agent config; added better device size and orientation logic; more commentary

bmcminn commented:
I like that idea. Just added those defs verbatim to `src/profiles/devices.js`.

I also went ahead and fixed up a few other things and updated the PR description. I think this feature is ready to close now.

shyiko commented:
Hey man. Unfortunately it looks like this PR will have to wait. My laptop won't turn on :) Will get back to it as soon as I have it fixed.

bmcminn commented:
Hey @shyiko any update on your laptop situation?

shyiko commented:
@bmcminn I'm getting it back tomorrow morning.
